src/hypothesis/internal/conjecture/data.py

```diff
@@ -100,14 +106,28 @@ def note(self, value):
         self.output += value
 
     def draw(self, strategy):
-        if self.depth >= MAX_DEPTH:
-            self.mark_invalid()
-
         if self.is_find and not strategy.supports_find:
             raise InvalidArgument((
                 'Cannot use strategy %r within a call to find (presumably '
                 'because it would be invalid after the call had ended).'
             ) % (strategy,))
+
+        if self.depth >= MAX_DEPTH:
+            self.mark_invalid()
+        if self.depth == 0 and not IN_COVERAGE_TESTS:  # pragma: no cover
+            original_tracer = sys.gettrace()
+            if original_tracer is None:
```

Comment:
It's safe, but I actually think (and I'm pretty sure this *wasn't* what I was thinking when I wrote this, but it's still a valid point) that there's a non-trivial difference under pypy because settrace is opaque to the JIT. The code-path for the C code for `settrace(None)` on CPython isn't totally trivial either.

Reply:
A combination of talking to the pypy folks and realising that this path is never hit on recursive calls to draw() has convinced me you're right. I'll take it out.